R/tm_a_pca.R

```diff
@@ -224,7 +239,8 @@ ui_a_pca <- function(id, ...) {
               label = "Plot type",
               choices = args$plot_choices,
               selected = args$plot_choices[1]
-            )
+            ),
+            ui_teal_transform_data(ns("decorate"), transformators = args$decorators)
```

Comment:
This makes more sense in line 261, at the top of the panel labelled `title = "Selected plot specific settings",`

Reply:
Example app:

```r
# PCA

devtools::load_all("../teal")
devtools::load_all(".")
plot_title <- teal_transform_module(
  server = make_teal_transform_server(expression({
    plot <- plot + ggtitle("Custom title")
  }))
)

interactive_decorator <- teal_transform_module(
  ui = function(id) {
    ns <- NS(id)
    div(
      textInput(ns("x_axis_title"), "X axis title", value = "x axis")
    )
  },
  server = function(id, data) {
    moduleServer(id, function(input, output, session) {
      reactive({
        req(data())
        within(data(),
               {
                 plot <- plot +
                   xlab(my_title)
               },
               my_title = input$x_axis_title
        )
      })
    })
  }
)

# general data example
data <- teal_data()
data <- within(data, {
  require(nestcolor)
  USArrests <- USArrests
})

app <- init(
  data = data,
  modules = modules(
    tm_a_pca(
      "PCA",
      dat = data_extract_spec(
        dataname = "USArrests",
        select = select_spec(
          choices = variable_choices(
            data = data[["USArrests"]], c("Murder", "Assault", "UrbanPop", "Rape")
          ),
          selected = c("Murder", "Assault"),
          multiple = TRUE
        ),
        filter = NULL
      ),
      decorators = list(plot_title, interactive_decorator)
    )
  )
)
if (interactive()) {
  shinyApp(app$ui, app$server)
}
```

Reply:
![image](https://github.com/user-attachments/assets/3c9488e3-a9b6-48b6-987b-074096955b78)

Reply:
thanks @averissimo - I don't want to hide decorators UI for now and this seems like a bit hidden